admin_app/src/app/integrations/page.tsx

```diff
@@ -43,11 +45,33 @@ const KeyManagement = ({
   token: string | null;
   editAccess: boolean;
 }) => {
-  // const [currentKey, setCurrentKey] = useState("qwerty...");
+  const [keyInfoFetchIsLoading, setKeyInfoFetchIsLoading] = useState(true);
+  const [currentKey, setCurrentKey] = useState("");
+  const [currentKeyLastUpdated, setCurrentKeyLastUpdated] = useState("");
   const [confirmationModalOpen, setConfirmationModalOpen] = useState(false);
   const [newKeyModalOpen, setNewKeyModalOpen] = useState(false);
   const [newKey, setNewKey] = useState("");
-  const [isLoading, setIsLoading] = useState(false);
+  const [keyGenerationIsLoading, setKeyGenerationIsLoading] = useState(false);
+
+  React.useEffect(() => {
+    const setApiKeyInfo = async () => {
+      setKeyInfoFetchIsLoading(true);
+      try {
+        const data = await apiCalls.getUser(token!);
+        setCurrentKey(data.api_key_first_characters);
+        const formatted_api_update_date = format(
+          data.api_key_updated_datetime_utc,
+          "HH:mm, dd-MM-yyyy",
+        );
+        setCurrentKeyLastUpdated(formatted_api_update_date);
+        setKeyInfoFetchIsLoading(false);
+      } catch (error) {
+        console.error(error);
+        setKeyInfoFetchIsLoading(false);
+      }
+    };
+    setApiKeyInfo();
+  }, [token, newKey]);
```

Comment:
Do we need a fix like what Suzin added to not throw an error here since we are calling the api?

Reply:
Didn't see an issue pop up, but added the check as a precaution anyway